### PR TITLE
#821 Fix accessibility issues related to lists and list items

### PR DESCRIPTION
**Pull request by katarinazaprazna**
- Ensure `NavigationMenu.List` has only `NavigationMenu.Item` components as direct children

Screenshot 2024-11-04 at 11 34 27

### CODE REVIEW

**Review by radoslavzeman — COMMENTED** [review]
Could we somehow keep the figma design with search on the right while implementing the Navigation menu correctly?

File: next/src/components/common/NavBar/NavMenu/NavMenu.tsx [hunk 1]
```diff
@@ -44,19 +44,20 @@ const NavMenu = ({ className }: NavMenuProps) => {
...
-            {menus?.map((menuItem) => <NavMenuItem key={menuItem.id} menuItem={menuItem} />)}
-          </div>
-          <Button
-            href={getLinkProps(searchLink).href}
-            asLink
-            icon={<Icon name="lupa" className="size-6" />}
-            hasLinkIcon={false}
-            aria-label={t('navBar.aria.searchButton')}
-            variant="icon-wrapped"
-            className="-mr-4 px-4 py-5"
```
Comment (radoslavzeman) [inline next/src/components/common/NavBar/NavMenu/NavMenu.tsx:58]: ..please keep `-mr-4` to align the icon with container..

**Review by radoslavzeman — APPROVED** [review]
👍

File: next/src/components/common/NavBar/NavMenu/NavMenu.tsx (hunk 1 above)
Comment (katarinazaprazna) [reply next/src/components/common/NavBar/NavMenu/NavMenu.tsx:58]: I now see the use case for `-mr-4` - adding it back

**Pull request merged**